src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -403,6 +415,8 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
             }
 
             ProcessCodeDirect();
+             //Recreate connectors that can be reused
+             LoadAndCreateConnectors(inportConnections, outportConnections);
```

Comment:
To me it means any changes made to a node with connector will make the graph no longer identical. It sounds bad to me but I don't see a good example of graph comparison evaluating connections right now. Did you check what's the behavior for node delete and undo, I feel we better align those two

Reply:
I'll check that, my guess is that is recreates the same node id as the whole node is serialized to xml. Thats not the case with the ports or connectors as we're creating brand new ones. I think we can figure out how to manually set the ID back to the original ID, it's just more work for I'm not sure how much gain.

Reply:
That's my first thinking as well. I don't know the compute process enough to say if we maintain the connector id, does it help reduce engine compute work or graph comparison accuracy

Reply:
yeah, let me see how much work it is, the test is already written.

Reply:
after quick discussion @QilongTang and I decided we would handle this as a separate task, as the initial bug will be hard enough to test. 
tracked:
https://jira.autodesk.com/browse/QNTM-3197